tests/llmobs/test_llmobs.py

```diff
@@ -98,6 +102,157 @@ def test_propagates_ignore_non_llmobs_spans(self, tracer, llmobs_events):
+        env["DD_LLMOBS_ENABLED"] = "1"
+        env["DD_LLMOBS_AGENTLESS_ENABLED"] = "0"
+        env["DD_TRACE_ENABLED"] = "0"
+        env["DD_TRACE_AGENT_URL"] = llmobs_backend.url()
+        out, err, status, _ = ddtrace_run_python_code_in_subprocess(
+            dedent(
+                """
+            from ddtrace.llmobs import LLMObs
+
+            def span_processor(s):
+                if s.get_tag("scrub_values") == "1":
+                    s.input_messages = [{"content": "scrubbed"}]
+                    s.output_messages = [{"content": "scrubbed"}]
+                return s
+
+            LLMObs.register_processor(span_processor)
+
+            with LLMObs.llm("openai.request") as llm_span:
+                LLMObs.annotate(llm_span, input_data="value", output_data="value", tags={"scrub_values": "1"})
+            with LLMObs.llm("openai.request") as llm_span:
+                LLMObs.annotate(llm_span, input_data="value", output_data="value", tags={"scrub_values": "0"})
+            """
+            ),
+            env=env,
+        )
+        assert out == b""
+        assert status == 0, err
+        assert err.decode() == ""
+        events = llmobs_backend.wait_for_num_events(num=1)
+        traces = events[0]
+        assert len(traces) == 2
+        assert "scrub_values:1" in traces[0]["spans"][0]["tags"]
+        assert traces[0]["spans"][0]["meta"]["input"]["messages"][0]["content"] == "scrubbed"
+        assert traces[0]["spans"][0]["meta"]["output"]["messages"][0]["content"] == "scrubbed"
+        assert "scrub_values:0" in traces[1]["spans"][0]["tags"]
+        assert traces[1]["spans"][0]["meta"]["input"]["messages"][0]["content"] == "value"
+        assert traces[1]["spans"][0]["meta"]["output"]["messages"][0]["content"] == "value"
+
+    def test_register_unregister_processor(self, llmobs, llmobs_events):
+        def _sp(s):
+            s.input_messages = [{"content": "scrubbed"}]
+            return s
+
+        # register
+        llmobs.register_processor(_sp)
+        with llmobs.llm("openai.request") as llm_span:
+            llmobs.annotate(llm_span, input_data="value")
+        assert llmobs_events[0]["meta"]["input"]["messages"][0]["content"] == "scrubbed"
+
+        # unregister
+        llmobs.register_processor(None)
+        with llmobs.llm("openai.request") as llm_span:
+            llmobs.annotate(llm_span, input_data="value")
+        assert llmobs_events[1]["meta"]["input"]["messages"][0]["content"] == "value"
+
+    def test_processor_error_is_logged(self, ddtrace_run_python_code_in_subprocess, llmobs_backend):
+        """Ensure that when an exception is raised an exception is logged."""
+        env = os.environ.copy()
+        env["DD_LLMOBS_ML_APP"] = "test-ml-app"
+        env["DD_API_KEY"] = "test-api-key"
+        env["DD_LLMOBS_AGENTLESS_ENABLED"] = "0"
+        env["DD_TRACE_ENABLED"] = "0"
+        env["DD_TRACE_AGENT_URL"] = llmobs_backend.url()
+        env["DD_TRACE_LOGGING_RATE"] = "0"
```

Comment:
took me way too long to learn that

> 1 log record per name/level/pathname/lineno every 60 seconds by default

is the default logging configuration for the library 🙂